crates/core/src/store.rs

```diff
@@ -264,7 +269,7 @@ impl StoreBuilder {
                 ctx.set_stderr(Box::new(wasi_preview1::pipe::WritePipe::new(w)))
             }
             WasiCtxBuilder::Preview2(ctx) => {
-                ctx.stderr(MyStdoutStream(Mutex::new(Some(Box::new(
+                ctx.stderr(MyStdoutStream(Arc::new(Mutex::new(Box::new(
```

Comment:
I think we need to make this a `tokio::sync::Mutex` instead of a `std::sync::Mutex` since we'll need to lock it during an `await` in the `Subscribe` implementation.